tests/Types/TypeRegistryTest.php

```diff
@@ -97,9 +97,11 @@ public function testRegisterWithAlreadyRegisteredInstance(): void
         $newType = new TextType();
 
         $this->registry->register('some', $newType);
-
-        $this->expectException(Exception::class);
-        $this->registry->register('other', $newType);
+        $this->registry->register('sameButDifferent', $newType);
+        self::assertSame(
+            $this->registry->get('some'),
+            $this->registry->get('sameButDifferent')
```

Comment:
"sameButDifferent" is really confusing to read. Why not just use some arbitrary keys like "type1" and "type2"? The fact that they are represented by the same object will be clear from the context and doesn't have to be expressed in the names.